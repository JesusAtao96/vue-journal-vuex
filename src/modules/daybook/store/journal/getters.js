/* eslint-disable no-unused-vars */
export const getEntriesByTerm = (state) => {
    
}

export const getEntryById = (state) => {
    
}