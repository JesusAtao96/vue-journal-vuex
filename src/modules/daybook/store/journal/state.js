export default () => ({
    isLoading: true,
    entries: [
        {
            id: new Date().getTime(),
            date: new Date().toDateString(),
            text: 'Ipsum id do velit amet dolor do fugiat elit elit excepteur magna incididunt aliqua. Nostrud irure esse quis officia enim deserunt laboris. Proident aliqua pariatur in esse in deserunt ad id aute non cupidatat amet adipisicing cupidatat.',
            picture: null,
        },
        {
            id: new Date().getTime() + 1000,
            date: new Date().toDateString(),
            text: 'Voluptate nisi deserunt labore in ad excepteur occaecat reprehenderit proident culpa. Cupidatat elit id id elit dolore laboris et veniam ut ut est est esse velit. Eiusmod esse excepteur laborum dolore ipsum esse reprehenderit tempor tempor adipisicing nisi. Sit minim elit tempor cupidatat nulla pariatur occaecat ex. Fugiat reprehenderit non excepteur ad elit irure Lorem. Sint mollit aliqua Lorem elit cupidatat. Quis sint minim exercitation minim elit eiusmod pariatur velit quis.',
            picture: null,
        },
        {
            id: new Date().getTime() + 2000,
            date: new Date().toDateString(),
            text: 'Minim est id anim non dolor. Culpa sunt enim dolore esse cupidatat enim sint elit aute consectetur. Cillum amet proident ex magna irure qui ut sint dolore ullamco mollit reprehenderit consequat.',
            picture: null,
        },
    ],
})