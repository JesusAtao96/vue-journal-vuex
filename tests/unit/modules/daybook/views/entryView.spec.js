import { createStore } from 'vuex'
import { shallowMount } from "@vue/test-utils";

import Swal from "sweetalert2";

import EntryView from "@/modules/daybook/views/EntryView";
import journal from '@/modules/daybook/store/journal'

import { journalState } from "../../../mock-data/test-journal-state";

const createVuexStore = (initialState) => createStore({
    modules: {
        journal: {
            ...journal,
            state: { ...initialState }
        }
    }
})

jest.mock('sweetalert2', () => ({
    fire: jest.fn(),
    showLoading: jest.fn(),
    close: jest.fn(),
}))

describe("Pruebas en el EntryView", () => {
    const store = createVuexStore(journalState)
    store.dispatch = jest.fn()

    const mockRouter = {  push: jest.fn() };

    let wrapper

    beforeEach(() => {
        jest.clearAllMocks()
        wrapper = shallowMount(EntryView, {
            props: {
                id: '-MhvKCjNX5oZI8ghNmrc'
            },
            global: {
                mocks: {
                    $router: mockRouter
                },
                plugins: [ store ]
            }
        })
    })

    test("debe de sacar al usuario por que el id no existe", () => {
        // eslint-disable-next-line no-unused-vars
        const wrapper = shallowMount(EntryView, {
            props: {
                id: 'Este ID no exite en el store'
            },
            global: {
                mocks: {
                    $router: mockRouter
                },
                plugins: [ store ]
            }
        })

        expect(mockRouter.push).toHaveBeenCalledWith({ name: 'no-entry' })
    });

    test("debe de mostrar la entrada correctamente", () => {
        expect(wrapper.html()).toMatchSnapshot()
        expect(mockRouter.push).not.toHaveBeenCalled()
    });

    test("debe de borrar la entrada y salir", (done) => {
        Swal.fire.mockReturnValueOnce(Promise.resolve({ isConfirmed: true }))
        wrapper.find('.btn-danger').trigger('click')

        expect(Swal.fire).toHaveBeenCalledWith({
            title: '¿Está seguro?',
            text: 'Una vez borrado, no se puede recuperar',
            showDenyButton: true,
            confirmButtonText: 'Si, estoy seguro'
        })

        setTimeout(() => {
            expect(store.dispatch).toHaveBeenCalledWith('journal/deleteEntry', '-MhvKCjNX5oZI8ghNmrc')
            expect(mockRouter.push).toHaveBeenCalled()
            done()
        }, 1)
    });
});