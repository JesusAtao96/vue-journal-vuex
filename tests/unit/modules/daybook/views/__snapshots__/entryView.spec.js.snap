// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el EntryView debe de mostrar la entrada correctamente 1`] = `
<div class="entry-title d-flex justify-content-between p-2">
  <div><span class="text-success fs-3 fw-bold">24</span><span class="mx-1 fs-3">Agosto</span><span class="mx-2 fs-4 fw-light">2021, Martes</span></div>
  <div><input type="file" accept="image/png, image/jpeg" style="display: none;"><button class="btn btn-danger mx-2">Borrar <i class="fas fa-trash-alt"></i></button><button class="btn btn-primary mx-2">Subir Foto <i class="fas fa-upload"></i></button></div>
</div>
<hr>
<div class="d-flex flex-column px-3 h-75"><textarea class="form-control" placeholder="¿Qué sucedió hoy?"></textarea></div>
<fab-stub icon="fa-save"></fab-stub><img src="https://res.cloudinary.com/jealeat96/image/upload/v1629929589/kacbr99lcmq5b5a5blff.jpg" alt="entry-picture" class="img-thumbnail">
<!--v-if-->
`;
